x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,306 @@
+	}{
+		{
+			name: "set new authorization successfully",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			addAuthorization: types.Authorization{
+				MsgUrl:           "XYZ",
+				AuthorizedPolicy: types.PolicyType_groupOperational,
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+				{
+					MsgUrl:           "XYZ",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+		},
+		{
+			name: "update existing authorization successfully",
+			oldList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			addAuthorization: types.Authorization{
+				MsgUrl:           "ABC",
+				AuthorizedPolicy: types.PolicyType_groupEmergency,
+			},
+			expectedList: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupEmergency,
+				},
+			}},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.oldList.SetAuthorizations(tc.addAuthorization)
+			require.Equal(t, tc.expectedList, tc.oldList)
+		})
+	}
+}
+
+func TestAuthorizationList_GetAuthorizations(t *testing.T) {
+	tt := []struct {
+		name            string
+		authorizations  types.AuthorizationList
+		getPolicyMsgUrl string
+		expectedPolicy  types.PolicyType
+		error           error
+	}{
+		{
+			name: "get authorizations successfully",
+			authorizations: types.AuthorizationList{Authorizations: []types.Authorization{
+				{
+					MsgUrl:           "ABC",
+					AuthorizedPolicy: types.PolicyType_groupOperational,
+				},
+			}},
+			getPolicyMsgUrl: "ABC",
+			expectedPolicy:  types.PolicyType_groupOperational,
+			error:           nil,
+		},
+		{
+			name:            "get authorizations fails when msg not found in list",
+			authorizations:  types.AuthorizationList{Authorizations: []types.Authorization{}},
+			getPolicyMsgUrl: "ABC",
+			expectedPolicy:  types.PolicyType(0),
+			error:           types.ErrAuthorizationNotFound,
+		},
+		{
+			name:            "get authorizations fails when when queried for empty string",
+			authorizations:  types.AuthorizationList{Authorizations: []types.Authorization{}},
+			getPolicyMsgUrl: "",
+			expectedPolicy:  types.PolicyType(0),
+			error:           types.ErrAuthorizationNotFound,
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			policy, err := tc.authorizations.GetAuthorizedPolicy(tc.getPolicyMsgUrl)
+			require.ErrorIs(t, err, tc.error)
+			require.Equal(t, tc.expectedPolicy, policy)
+		})
+	}
+}
+
+func TestAuthorizationList_Validate(t *testing.T) {
```

Comment:
Need to check for empty list.

Also I think we can check Default list is valid as well. Here there is not a specific assertion to check, this is more like an integration test

Reply:
c18bd7d3776754598a3ebc3a527995e71faf7f15

What do you mean by any integration test ?

Reply:
By testing `DefaultAuthorizationList` we don't test the validate function itself but the integration between default list and validate function